packages/lib/src/vis/shared/AxisSystem.tsx

```diff
@@ -1,52 +1,55 @@
+import { assertDefined, assertNonNull } from '@h5web/shared';
+import { useThree } from '@react-three/fiber';
+import { createPortal } from 'react-dom';
+
 import { useCameraState } from '../hooks';
 import type { AxisOffsets } from '../models';
 import Axis from './Axis';
-import styles from './AxisSystem.module.css';
 import Html from './Html';
 import { useVisCanvasContext } from './VisCanvasProvider';
 
 interface Props {
   axisOffsets: AxisOffsets;
-  title?: string;
   showAxes: boolean;
 }
 
 function AxisSystem(props: Props) {
-  const { axisOffsets, title, showAxes } = props;
+  const { axisOffsets, showAxes } = props;
   const { canvasSize, abscissaConfig, ordinateConfig, getVisibleDomains } =
     useVisCanvasContext();
 
+  const visCanvas = useThree(
+    (state) => state.gl.domElement.parentElement?.parentElement?.parentElement,
+  );
```

Comment:
That seems a bit brittle. 

Could we imaging exposing a ref to `visCanvas` through the `VisCanvas` ?

Reply:
Good call. I'll do it in one of my later PRs to avoid rebasing conflicts.